components/DetailPageSidebar/DetailPageSidebar.js

```diff
@@ -0,0 +1,197 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import { Notification } from 'carbon-components';
+
+class DetailPageSidebar extends Component {
+    static propTypes = { 
+        notification: PropTypes.shape({
+            kind: PropTypes.string,
+            title: PropTypes.string,
+            message: PropTypes.string,
+        }), 
+        creditNotification: PropTypes.shape({
+            kind: PropTypes.string,
+            title: PropTypes.string,
+            message: PropTypes.string,
+        }),
+        artifact: PropTypes.shape({
+            name: PropTypes.string,
+            id: PropTypes.string,
+            tag: PropTypes.string,
+            tagName: PropTypes.string,
+            description: PropTypes.string,
+            stageTag: PropTypes.string,
+            stageTagName: PropTypes.string,
+            accountType: PropTypes.string,
+            accountTagName: PropTypes.string,
+            availabilityTagName: PropTypes.string,
+            type: PropTypes.string,
+            docURL: PropTypes.string,
+            termsUrl: PropTypes.string,
+            version: PropTypes.string,
+            author: PropTypes.string,
+            createdDate: PropTypes.string,
+            publishDate: PropTypes.string,
+            locationName: PropTypes.string,
+            regionName: PropTypes.string,
+        }),
+        i18n: PropTypes.shape({
+            deprecationWarning: PropTypes.string,
+            experimentalDesc: PropTypes.string,
+            viewDocs: PropTypes.string,
+            viewTerms: PropTypes.string,
+            author: PropTypes.string,
+            version: PropTypes.string,
+            createdDate: PropTypes.string,
+            publishDate: PropTypes.string,
+            type: PropTypes.string,
+            boilerplate: PropTypes.string,
+            runtime: PropTypes.string,
+            service: PropTypes.string,
+            location: PropTypes.string,
+            region: PropTypes.string,
+        }),
+        children: PropTypes.node };
+    
+ 
+    render() {
+        const {
+            children,
+            notification,
+            creditNotification,
+            i18n,
+            artifact,
+            ...other
```

Comment:
So the reason we use the `spread` operator here is to allow users of a component to apply any kind of prop handler (the normal use case is someone needs to add an onClick or a onKeyDown or something like that) toe the component. Important to that is that the `...other` then needs to be spread across the returned component. So I'd just take a look and see what possible use-cases the `other` object could contain and map it to one of the `div's` down below.